cmd/git/git.go

```diff
@@ -1026,167 +1028,76 @@ func (j *DSGit) SetParentCommitFlag(richItem map[string]interface{}) (err error)
 }
 
 // GetModelData - return data in swagger format
-func (j *DSGit) GetModelData(ctx *shared.Ctx, docs []interface{}) (data *models.Data) {
-	data = &models.Data{
-		DataSource: GitDataSource,
-		MetaData:   gGitMetaData,
-		Endpoint:   j.URL,
-		RepositoryStats: &models.RepositoryStats{
-			CalculatedAt:              strfmt.DateTime(j.StatsDt),
-			ProgrammingLanguagesStats: j.ModelPLS,
-			RepositoryURL:             j.URL,
-			TotalLinesOfCode:          int64(j.Loc),
-		},
-	}
-	source := data.DataSource.Slug
+func (j *DSGit) GetModelData(ctx *shared.Ctx, docs []interface{}) []insights.Commit {
+	data := make([]insights.Commit, 0)
+	source := GitDataSource
 	for _, iDoc := range docs {
-		var (
-			message *string
-			title   *string
-		)
+		commit := insights.Commit{}
+		commit.Connector = "git"
+		commit.ConnectorVersion = GitBackendVersion
 		doc, _ := iDoc.(map[string]interface{})
-		docUUID, _ := doc["uuid"].(string)
-		commitURL, _ := doc["commit_url"].(string)
-		commitRepoType, _ := doc["commit_repo_type"].(string)
-		sha, _ := doc["hash"].(string)
-		hashShort, _ := doc["hash_short"].(string)
-		sMessage, _ := doc["message"].(string)
-		sTitle, _ := doc["title"].(string)
-		repoShortURL, _ := doc["repo_short_name"].(string)
-		docCommit, _ := doc["doc_commit"].(bool)
-		emptyCommit, _ := doc["empty_commit"].(bool)
-		locAdded, _ := doc["lines_added"].(int64)
-		locRemoved, _ := doc["lines_removed"].(int64)
-		locChanged, _ := doc["lines_changed"].(int64)
-		_, squashedCommit := j.OrphanedMap["sha"]
-		if sMessage != "" {
-			message = &sMessage
-		}
-		if sTitle != "" {
-			title = &sTitle
-		}
-		parents, _ := doc["parents"].([]string)
-		authoredInTz, _ := doc["author_date"].(time.Time)
+		commit.URL, _ = doc["commit_url"].(string)
+		commit.SHA, _ = doc["hash"].(string)
+		commit.ShortHash, _ = doc["hash_short"].(string)
+		commit.Source = doc["commit_repo_type"].(string)
+		commit.Message, _ = doc["message"].(string)
+		_, commit.MergeCommit = j.OrphanedMap["sha"]
```

Comment:
`_, commit.MergeCommit = j.OrphanedMap["sha"]` - this returns information if a given commit is orphaned or not (I've used this for `squashed` property) - I'm not sure if that gives you info about commit being a merge commit or not.

Reply:
this mapping is wrong, let me update it.

Reply:
updated.